lib/active_merchant/billing/gateways/agms.rb

```diff
@@ -0,0 +1,322 @@
+        # Call the commit function to pass on the params to gateway
+        commit('sale', post)
+      end
+
+      def authorize(money, payment, options={})
+        # Authorize transaction, same as purchase only we do not charge the card
+        post = {}
+        # CC Data, Check Data
+        add_payment(post, payment)
+        # Invoice Data
+        add_invoice(post, money, options)
+        # Address, Shipping Address Data
+        add_address(post, options)
+        # Customer Data (IP, EMail)
+        add_customer_data(post, options)
+        # Custom Data
+        add_custom_data(post, options)
+        # Call the commit function to pass on the params to gateway
+        commit('auth', post)
+      end
+
+      def capture(money, authorization, options={})
+        # Capture transaction, only payment and customer address is required
+        post = {}
+        # Invoice Data
+        add_invoice(post, money, options)
+        # Add Authorization
+        add_authorization(post, authorization, options)
+        # Call the commit function to pass on the params to gateway
+        commit('capture', post)
+      end
+
+      def refund(money, authorization, options={})
+        # Refund transaction, we need to pass on the authorization number along with amount for refund to locate previous trnsaction
+        post = {}
+        # Invoice Data
+        add_invoice(post, money, options)
+        # Add Authorization
+        add_authorization(post, authorization, options)
+        # Call the commit function to pass on the params to gateway
+        commit('refund', post)
+      end
+
+      def void(authorization, options={})
+        # Void transaction, we need to pass on only authorization number for voiding transaction
+        post = {}
+        # Add Authorization
+        add_authorization(post, authorization, options)
+        # Call the commit function to pass on the params to gateway
+        commit('void', post)
+      end
+
+      def verify(credit_card, options={})
+        # Verify is two step process, first we fire the authorize then we capture the auth and issue void
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def supports_scrubbing?
+        # Indicates whether before dumping the params it remove the sensitive information
+        true
+      end
+
+      def scrub(transcript)
+        # Scrubbing logic, we remove CCNumber, CVV, GatewayUserName and GatewayPassword before it is dumped
+        # The module also support dumping of all the request response in file for debugging and stubbing, this ensure no sensitive info is dumped
+        transcript.
+          gsub(%r((<CCNumber>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<CVV>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<GatewayUserName>)[^<]+(<))i, '\1[FILTERED]\2').
```

Comment:
Don't scrub non-sensitive credentials like user name since they're super useful for debugging.